src/mapper/mapper_test.go

```diff
@@ -38,3 +43,113 @@ func createNamespace(name string, labels map[string]string) *corev1.Namespace {
+	})
+	t.Run("Remove stale dynakube entry for no longer matching ns", func(t *testing.T) {
+		labels := map[string]string{"test": "selector"}
+		movedDk := createTestDynakubeWithAppInject("moved-dk", labels, nil)
+		nsLabels := map[string]string{
+			dtwebhook.InjectionInstanceLabel: movedDk.Name,
+		}
+		namespace := createNamespace("test-namespace", nsLabels)
+
+		updated, err := updateNamespace(namespace, &dynatracev1beta1.DynaKubeList{Items: []dynatracev1beta1.DynaKube{*movedDk}})
+
+		require.NoError(t, err)
+		require.True(t, updated)
+		assert.Equal(t, 0, len(namespace.Labels))
+	})
+	t.Run("Throw error in case of conflicting Dynakubes", func(t *testing.T) {
+		labels := map[string]string{"test": "selector"}
+		dk := createTestDynakubeWithMultipleFeatures("dk-test", labels, nil)
+		conflictingDk := createTestDynakubeWithMultipleFeatures("conflicting-dk", labels, nil)
+		nsLabels := map[string]string{
+			dtwebhook.InjectionInstanceLabel: dk.Name,
+			"test":                           "selector",
+		}
+		namespace := createNamespace("test-namespace", nsLabels)
+
+		_, err := updateNamespace(namespace, &dynatracev1beta1.DynaKubeList{Items: []dynatracev1beta1.DynaKube{*conflictingDk, *dk}})
+
+		assert.Error(t, err)
+	})
+	t.Run("Ignore kube namespaces", func(t *testing.T) {
+		dk := createTestDynakubeWithMultipleFeatures("appMonitoring", nil, nil)
+		namespace := createNamespace("kube-something", nil)
+
+		updated, err := updateNamespace(namespace, &dynatracev1beta1.DynaKubeList{Items: []dynatracev1beta1.DynaKube{*dk}})
+
+		require.NoError(t, err)
+		require.False(t, updated)
+		assert.Equal(t, 0, len(namespace.Labels))
+	})
+
+	t.Run("Ignore openshift namespaces", func(t *testing.T) {
+		dk := createTestDynakubeWithMultipleFeatures("appMonitoring", nil, nil)
+		namespace := createNamespace("openshift-something", nil)
+
+		updated, err := updateNamespace(namespace, &dynatracev1beta1.DynaKubeList{Items: []dynatracev1beta1.DynaKube{*dk}})
+
+		require.NoError(t, err)
+		require.False(t, updated)
+		assert.Equal(t, 0, len(namespace.Labels))
+	})
+	t.Run("Double dynakube, 1. ignores openshift namespaces, 2. doesn't", func(t *testing.T) {
+		labels := map[string]string{"test": "selector"}
+		otherLabels := map[string]string{"test1": "selector"}
+		ignoreDk := createTestDynakubeWithMultipleFeatures("appMonitoring", otherLabels, nil)
+		notIgnoreDk := createTestDynakubeWithMultipleFeatures("boom", labels, nil)
+		notIgnoreDk.Annotations = map[string]string{
+			dynatracev1beta1.AnnotationFeatureIgnoredNamespaces: "[\"asd\"]",
+		}
+		namespace := createNamespace("openshift-something", labels)
+
+		updated, err := updateNamespace(namespace, &dynatracev1beta1.DynaKubeList{Items: []dynatracev1beta1.DynaKube{*ignoreDk, *notIgnoreDk}})
+
+		require.NoError(t, err)
+		require.True(t, updated)
+		assert.Equal(t, 2, len(namespace.Labels))
+	})
+	t.Run("Double dynakube, 1. doesn't, 2. ignores openshift namespaces", func(t *testing.T) {
```

Comment:
this should check that the label has the correct dynakube assigned

Reply:
done